Classes/common/touchdb/TD_Database.m

```diff
@@ -367,7 +367,33 @@ INSERT INTO INFO (key, value) VALUES ('publicUUID',  '%@')",
                 result = NO;
                 return;
             }
-            // dbVersion = 6;
+            dbVersion = 6;
+        }
+        
+        if (dbVersion < 7) {
+            // Version 7: upgrade replicators.last_sequence to a json dict of {"seq": <data>}
+            FMResultSet *replicators = [db executeQuery:@"SELECT rowid, last_sequence FROM replicators"];
+            
+            while ([replicators next]) {
+                NSUInteger rowid = [replicators longForColumn:@"rowid"];
+                NSData *lastSequence = [replicators dataForColumn:@"last_sequence"];
+                NSDictionary *lastSequenceJsonMaybe = [TDJSON JSONObjectWithData:lastSequence options:0 error:nil];
+                if (lastSequenceJsonMaybe == nil) {
+                    // sequence not in json format, so upgrade it
+                    NSString *lastSequenceString = [replicators stringForColumn:@"last_sequence"];
+                    // the sequence might have been a number, or a string "<number-base64>"
+                    NSDictionary *dict = @{@"seq": lastSequenceString};
+                    NSData *lastSequenceJson = [TDJSON dataWithJSONObject:dict options:0 error:nil];
+                    [db executeUpdate:@"UPDATE replicators SET last_sequence=? WHERE rowid=?", lastSequenceJson, @(rowid)];
+                }
+            }
```

Comment:
I think we might need to upgrade the hundreds place, [as detailed at the top of the method](https://github.com/cloudant/CDTDatastore/blob/43944-couchdb-20-clocal-compat/Classes/common/touchdb/TD_Database.m#L247), for breaking changes to the schema as we're changing the way `last_sequence` is stored to something that older versions of the library will unpack incorrectly?